scopesim/effects/spectral_trace_list_utils.py

```diff
@@ -284,6 +273,125 @@ def map_spectra_to_focal_plane(self, fov):
+           the header of the hdulist, but this is not yet provided by scopesim
+        """
+        logging.info("Rectifying %s", self.trace_id)
+
+        wave_min = kwargs.get("wave_min",
+                              self.wave_min)
+        wave_max = kwargs.get("wave_max",
+                              self.wave_max)
+        if wave_max < self.wave_min or wave_min > self.wave_max:
+            logging.info("   Outside filter range")
+            return None
+        wave_min = max(wave_min, self.wave_min)
+        wave_max = min(wave_max, self.wave_max)
+        logging.info("   %.02f .. %.02f um", wave_min, wave_max)
+
+        # bin_width is taken as the minimum dispersion of the trace
+        bin_width = kwargs.get("bin_width", None)
+        if bin_width is None:
+            self._set_dispersion(wave_min, wave_max)
+            bin_width = np.abs(self.dlam_per_pix.y).min()
+        logging.info("   Bin width %.02g um", bin_width)
+
+        pixscale = from_currsys(self.meta['pixel_scale'])
+
+        # Temporary solution to get slit length
+        xi_min = kwargs.get("xi_min", None)
+        if xi_min is None:
+            try:
+                xi_min = hdulist[0].header["HIERARCH INS SLIT XIMIN"]
+            except KeyError:
+                logging.error("xi_min not found")
+                return None
+        xi_max = kwargs.get("xi_max", None)
+        if xi_max is None:
+            try:
+                xi_max = hdulist[0].header["HIERARCH INS SLIT XIMAX"]
+            except KeyError:
+                logging.error("xi_max not found")
+                return None
+
+        if wcs is None:
+            wcs = WCS(naxis=2)
+            wcs.wcs.ctype = ['WAVE', 'LINEAR']
+            wcs.wcs.cunit = ['um', 'arcsec']
+            wcs.wcs.crpix = [1, 1]
+            wcs.wcs.cdelt = [bin_width, pixscale] # PIXSCALE
+
+        # crval set to wave_min to catch explicitely set value
+        wcs.wcs.crval = [wave_min, xi_min]   # XIMIN
+
+        nlam = int((wave_max - wave_min) / bin_width) + 1
+        nxi = int((xi_max - xi_min) / pixscale) + 1
+
+        # Create interpolation functions if not provided
+        if interps is None:
+            logging.info("Computing image interpolations")
+            interps = make_image_interpolations(hdulist, kx=1, ky=1)
+
+        # Create Xi, Lam images (do I need Iarr and Jarr or can I build Xi, Lam directly?)
+        Iarr, Jarr = np.meshgrid(np.arange(nlam, dtype=np.float32),
+                                 np.arange(nxi, dtype=np.float32))
+        Lam, Xi = wcs.all_pix2world(Iarr, Jarr, 0)
+
+        # Make sure that we do have microns
+        Lam = Lam * u.Unit(wcs.wcs.cunit[0]).to(u.um)
+
+        # Convert Xi, Lam to focal plane units
+        Xarr = self.xilam2x(Xi, Lam)
+        Yarr = self.xilam2y(Xi, Lam)
+
+        rect_spec = np.zeros_like(Xarr, dtype=np.float32)
+
+        ihdu = 0
+        for hdu in hdulist:
```

Comment:
```suggestion
        for ihdu, hdu in enumerate(hdulist):
```

Reply:
That suggestion does not work like this - the reason is that `ihdu` is used to index the `interps` list, which is not exactly parallel to `hdulist` (the primary du of `hdulist` is jumped over). `enumerate(hdulist[1:])` should work for all existing cases, but I wouldn't want to rely on that. 
I had to rerun the notebook  to realise this, which is good justification for including it in automatic testing.

Reply:
```suggestion
        imghdulist = [hdu for hdu in hdulist if isinstance(hdu, fits.ImageHDU)]
        assert len(imhdulist) == len(interps)
        # from Python 3.10 on, the above assert statement could be replaced by passing
        #     strict=True to the zip below
        for hdu, interp in zip(imghdulist, interps):
```
As far as I can tell, the individual hdus are not modified in this loop, only accessed. Only drawback I could think of is memory consumption, because we essentially create a copy of `hdulist`. Either way, I don't want to overcomplicate things, and the original implementation is probably just fine.